tracer/src/Datadog.Trace/Activity/ActivityListener.cs

```diff
@@ -0,0 +1,511 @@
+                var activityCurrentDynMethod = new DynamicMethod("ActivityCurrent", typeof(object), Type.EmptyTypes, typeof(ActivityListener).Module, true);
+                var activityCurrentDynMethodIl = activityCurrentDynMethod.GetILGenerator();
+                activityCurrentDynMethodIl.EmitCall(OpCodes.Call, activityCurrentMethodInfo, null);
+                activityCurrentDynMethodIl.Emit(OpCodes.Ret);
+                _getCurrentActivity = (Func<object>)activityCurrentDynMethod.CreateDelegate(typeof(Func<object>));
+            }
+
+            void ChangeActivityDefaultFormat()
+            {
+                // We change the default ID format to W3C (so traceid and spanid gets populated)
+                if (Activator.CreateInstance(_activityType, string.Empty).TryDuckCast<IActivityFormat>(out var activityFormat))
+                {
+                    activityFormat.DefaultIdFormat = ActivityIdFormat.W3C;
+                }
+            }
+        }
+
+        private static void CreateActivityListenerInstance()
+        {
+            _activityListenerType = Type.GetType("System.Diagnostics.ActivityListener, System.Diagnostics.DiagnosticSource", throwOnError: true);
+            _activitySourceType = Type.GetType("System.Diagnostics.ActivitySource, System.Diagnostics.DiagnosticSource", throwOnError: true);
+            _activityCreationOptionsType = Type.GetType("System.Diagnostics.ActivityCreationOptions`1, System.Diagnostics.DiagnosticSource", throwOnError: true);
+            _activitySamplingResultType = Type.GetType("System.Diagnostics.ActivitySamplingResult, System.Diagnostics.DiagnosticSource", throwOnError: true);
+            _activityContextType = Type.GetType("System.Diagnostics.ActivityContext, System.Diagnostics.DiagnosticSource", throwOnError: true);
+            _sampleActivityType = Type.GetType("System.Diagnostics.SampleActivity`1, System.Diagnostics.DiagnosticSource", throwOnError: true);
+
+            _onActivityStartedMethodInfo = typeof(ActivityListenerHandler).GetMethod("OnActivityStarted", BindingFlags.Static | BindingFlags.Public);
+            _onActivityStoppedMethodInfo = typeof(ActivityListenerHandler).GetMethod("OnActivityStopped", BindingFlags.Static | BindingFlags.Public);
+            _onSampleMethodInfo = typeof(ActivityListenerHandler).GetMethod("OnSample", BindingFlags.Static | BindingFlags.Public);
+            _onSampleUsingParentIdMethodInfo = typeof(ActivityListenerHandler).GetMethod("OnSampleUsingParentId", BindingFlags.Static | BindingFlags.Public);
+
+            Log.Information("Activity listener: {activityListenerType}", _activityListenerType.AssemblyQualifiedName ?? "(null)");
+
+            // Create the ActivityListener instance
+            var activityListener = Activator.CreateInstance(_activityListenerType);
+            var activityListenerProxy = _activityListenerInstance.DuckCast<IActivityListener>();
```

Comment:
I think this is a bug from be584e5

```suggestion
            var activityListenerProxy = activityListener.DuckCast<IActivityListener>();
```

Reply:
true, fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2446/commits/9c00bdcbc3c6acb0412def01171646c53701314a